### DISCUSSION TITLE
Error: cannot open this file as a SpatRaster

### DISCUSSION THREAD

**Discussion by millie-robson** [RangeShifter/RangeshiftR-tutorials#39] [Q&A] [answered]
Hello,

I am trying to run a patch-based model using the RangeShifter platform in RStudio, but I am encountering a few errors when trying to load in my habitat patches.

The habitat patches were created in ArcMap and converted into raster format, then the attributes were exported as an asc. file. I am trying to run the following code to read in the habitat patches:

patches <- terra::rast(paste0(dirpath,'Inputs/Raster_MCE_Std2_10_largest_areas.asc'))

But I receive the following errors:
Error: [rast] cannot open this file as a SpatRaster: /Users/millierobson/Documents/UNI/MRes/RangeShifter/RangeShifter PM/Models/PM_Reintroduction/Inputs/Raster_MCE_Std2_10_largest_areas.asc
In addition: Warning message:
`/Users/millierobson/Documents/UNI/MRes/RangeShifter/RangeShifter PM/Models/PM_Reintroduction/Inputs/Raster_MCE_Std2_10_largest_areas.asc' not recognized as a supported file format. (GDAL error 4)

I have installed and loaded all the relevant packages (raster, terra, rgdal).

I do not understand the errors here and I was hoping someone may be able to help. I have attached the asc. file. I appreciate any help!

Many thanks.

**Comment by JetteReeg** [comment]
Hi, 
as I said in the corresponding issue #38 this problem is related to [terra](https://github.com/rspatial/terra/issues/) and I recommend to post your issue there. However, if you attach the file I could have a quick check whether there is something wrong with the file.

**Reply by millie-robson** [reply]
Hello,

Thank you so much for your response and your help. When I try to attach the file on this thread it says it does not support the asc. file type.  I have instead attached it as a txt file.


[Raster_MCE_Std2_10_largest_areas.txt](https://github.com/RangeShifter/RangeshiftR-tutorials/files/11157151/Raster_MCE_Std2_10_largest_areas.txt)

**Reply by JetteReeg** [reply, marked as answer]
Hi, 
looking at the content of your file it seems that the file format is wrong and not in a raster format. You probably need to go back to ArcMap and make sure to rasterize your landscape (based on the content of your file it might be a spatial polygon or vector?). 

A typical .asc raster file would have a header with 6 rows defining the parameters 'ncols', 'nrows', 'xllcorner', 'yllcorner', 'cellsize' and 'NODATA_value', then following the cell values for each grid cell. You can google 'asc raster file example' for examples.

**Reply by millie-robson** [reply]
Hi,

Thank you so much! That has solved it - my file on ArcGIS was in raster format but I needed to use the Raster to ASCII tool in ArcGIS rather than just exporting the attribute table as an asc. file. 

Thank you so much for your help, I really appreciate it!
